### DISCUSSION TITLE
POV Autorecord Demo

### DISCUSSION THREAD

**Discussion by Haco** [SmileYzn/PugMod-AMXX#38] [Ideas]
I know there are autorecord plugins out there and they are working. But it would be better when there is a built-in demo record for your Pug-Mod, cause you can handle everything based on the different stages of the PUG.

**Problems with the available Autorecord Plugins:**
* Will capture everything in a single .dem file, even if you are on the server and waiting for players.
* When the server changes the map, the demo will continue to record into the same .dem file. This causes playback issues with "viewdemo" because then demo will stop playing as soon as the server changes the map. So basically everything is lost after the server changes the map.

**With a built-in autorecorder you could:**
1. Start recording the demo when the actual game starts (directly after the Team Vote and before the knife round starts. So the knife round is recorded also)
2. Force stop recording demo at half-time
3. After swap teams start recording the second half into a NEW .dem file. (with different name, see below)
4. Important: when the game ends, force "stopdemo", so it won't continue to record when the server changes map

Things to consider:
* Introduce cvar to enable/disable autorecord
* Introduce cvar to customize the prefix
* Demoname could be: prefix-de_dust2_HALF1.dem & prefix-de_dust2_HALF2.dem
* Or prefix-de_dust2_CT.dem and *_T.dem
* When any recording starts, you could send an info message in saychat, that indicates the demorecord + filename. (e. g. "Now recording to 'prefix-de_dust2_CT.dem'")

**Comment by SmileYzn** [comment]
I did not know if server can force the execution of these commands in client?

**Reply by Haco** [reply]
You can check this as reference, there is also the .sma file
https://forums.alliedmods.net/showthread.php?p=1399575

sma
https://forums.alliedmods.net/attachment.php?attachmentid=83977&d=1300877260

amxx
https://www.amxmodx.org/plcompiler_vb.cgi?file_id=83977

`client_cmd(id, "stop; record ^"%s^"", DemoName)`

Maybe this helps

**Reply by SmileYzn** [reply]
I think the hlds can't send these commands in new versions of game.
